SpatialGDK/Source/SpatialGDKServices/Private/SpatialGDKServicesModule.cpp

```diff
@@ -137,16 +140,34 @@ void FSpatialGDKServicesModule::ExecuteAndReadOutput(const FString& Executable,
 	FPlatformProcess::ClosePipe(0, WritePipe);
 }
 
+void FSpatialGDKServicesModule::SetProjectName(const FString& InProjectName)
+{
+	FString SpatialFileName = TEXT(SpatialOSConfigFile);
+	FString SpatialFileResult;
+
+	JsonParsedSpatialFile->SetStringField("name", InProjectName);
```

Comment:
We should check this is a valid shared pointer.

Reply:
`JsonParsedSpatialFile` could be  a null ptr.  Added check logic here.